asf_core_data/pipeline/epc/enrich/installer_proximity_indicator_utils.py

```diff
@@ -0,0 +1,160 @@
+        elif company_name in list(set(cleaned_installer_companies.installer_name)):
+            central_point = tuple(
+                cleaned_installer_companies[
+                    cleaned_installer_companies["installer_name"] == company_name
+                ][["longitude", "latitude"]].values[0]
+            )
+
+        dists = [
+            distance(central_point, install_latlong) for install_latlong in latlongs
+        ]
+
+        if len(dists) > 1:
+            dist_25, dist_50, dist_75 = np.percentile(dists, [25, 50, 75], axis=0)
+            comp_buffers = [
+                generate_geodesic_point_buffer(
+                    central_point[1], central_point[0], dist.km
+                )
+                for dist in (min(dists), dist_25, dist_50, dist_75, max(dists))
+            ]
+            company_install_shapes[company_name] = {
+                "min_buffer": comp_buffers[0],
+                "25_buffer": comp_buffers[1],
+                "50_buffer": comp_buffers[2],
+                "75_buffer": comp_buffers[3],
+                "max_buffer": comp_buffers[4],
+            }
+        else:
+            comp_buffer = generate_geodesic_point_buffer(
+                central_point[1], central_point[0], dists[0].km
+            )
+            # tbd if this should be considered min or max
+            company_install_shapes[company_name] = {"min_buffer": comp_buffer}
+
+    return company_install_shapes
+
+
+def calculate_buffer_distribution(company_install_shapes, epc_geo):
+    """Calculates the count of company distance spread buffers per epc
+     building.
+
+    Inputs:
+        epc_geo (np.ndarray): The latitude and longitude of an epc_building.
+
+    Outputs:
+        Counter(buffer_distribution) (dict): company distance spread buffer counts
+    """
+    buffer_distribution = []
+    for company, company_shapes in company_install_shapes.items():
+        in_buffer = [shape.contains(epc_geo) for shape in company_shapes.values()]
+        if any(in_buffer) is not False:
+            smallest_buffer = list(company_shapes.keys())[in_buffer.index(True)]
+            buffer_distribution.append(smallest_buffer)
+    return Counter(buffer_distribution)
+
+
+def calculate_weighted_installer_proximity(buffer_distb: Counter) -> int:
+    """Calculates weighted installer closeness score based
+    on distribution of companies in the min, 25th, 50th,
+    75th and max buffer distances. Score is between 0 and 1, where
+    a 1 is a perfect "closeness" score.
+
+    Inputs:
+        buffer_dist (Counter): The count of companies in the min,
+        25th, 50th, 75th and max buffer distance in relation to the
+        EPC home.
+
+    Outputs:
+        weighted_installer_closeness_score (int): weighted installer closeness score.
+    """
+    weighted_installer_closeness_score = 0
+    for buffer_dist, buffer_count in buffer_distb.items():
+        if "min_" in buffer_dist:
```

Comment:
to cut down on ink, before the `if` you could set `buffer_count / buffer_distb["max_buffer"]` to a variable and then use it in the rest of the calculations